text/0000-deprecation.md

```diff
@@ -0,0 +1,243 @@
+to actually remove an API, because that would break code. Even minor
+breaking changes (as [RFC PR #1156](https://github.com/rust-lang/rfcs/pull/1156) 
+cited above) generate huge discussion, because they call the general
+stability of the language and environment into question.
+
+The problem with opt-out, as defined by 
+[RFC #1122](https://github.com/rust-lang/rfcs/blob/master/text/1122-language-semver.md)
+is that code which previously compiled without problems stops working
+on a Rust upgrade, and requires manual intervention to get working 
+again.
+
+This has the long-term potential of fragmenting the Rust ecosystem into
+crates/projects using certain Rust versions and thus should be avoided
+at some (but not all) cost.
+
+Note that a similar problem exists with deprecation as defined: People
+may get used to deprecation warnings or just turn them off until their
+build breaks. Worse, the current setup creates a mirror world for
+libraries, in which deprecation doesn't exist!
+
+## 2. User Experience
+
+The following motivates items 2, 4 and 5.
+
+Currently, there is no way to make an API item unavailable via 
+deprecation. This means the API will only ever expand, with a lot of
+churn (case in point, the Java language has about 20% deprecated API
+surface [citation needed]). To better lead users to the right APIs and
+to allow for more effective language/API evolution, this proposal adds
+the `removed_at="<version>"` item to the `#[deprecated]` attribute.
+
+This allows us to effectively remove an API item from a certain target
+version while avoiding breaking code written for older target versions.
+
+Also rustc can emit better error messages than it could were the API
+items actually removed. In the best case, the error messages can steer
+the user to a working replacement.
+
+We want to avoid users setting `#[allow(deprecate)]` on their code to
+get rid of the warnings. On the other hand, there have been instances
+of failing builds because code was marked with `#![deny(deprecate)]`,
+namely `compiletest.rs` and all crates using it. This shows that the
+current system has room for improvement.
+
+We want to avoid failing builds because of wrong or missing target
+version definition. Therefor supplying useful defaults is of the 
+essence.
+
+The documentation can be optionally reduced to items relating to the
+current target version (e.g. by a switch), or deprecated items 
+relegated to a separate space, to reduce clutter and possible user
+confusion.
+
+## 3. Security Considerations
+
+I believe that *should* a security issue in one of our APIs be found,
+a swift and effective response will be required, and the current rules
+make no provisions for it. Thus proposal item 3.
+
+# Detailed design
+
+Cargo parses the additional `rust = "..."` dependency as if it was a 
+library. The usual rules for version parsing apply. If no `rust` 
+dependency is supplied, it can either default to `*`.
```

Comment:
Yeah, it should originally read 'or 1.0.0', because that would keep code without a rust dependency from breaking. However, that may also introduce spurious warnings/errors, so I removed it.